src/lib/webSocketClient.ts

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable class-methods-use-this */
+import { v4 as genUuid } from 'uuid';
+import camelcaseKeysDeep from 'camelcase-keys-deep';
+import { IMiddlewareWebSocketSubscriptionMessage, WebSocketChannelName } from '../types';
+import {
+  WEB_SOCKET_CHANNELS,
+  WEB_SOCKET_SUBSCRIBE,
+  WEB_SOCKET_UN_SUBSCRIBE,
+  handleUnknownError,
+} from '../popup/utils';
+
+let wsClient: WebSocket;
+let isWsConnected: boolean;
+
+const subscribersQueue: IMiddlewareWebSocketSubscriptionMessage[] = [];
+
+const subscribers: Record<WebSocketChannelName, Record<string, (payload: any) => void>> = {
```

Comment:
`(payload: any) => void` is used many times in this file. Maybe consider creating a type for this, like for example `WebSocketCallback` or something similar.